kafka/src/main/java/io/specmesh/kafka/Provisioner.java

```diff
@@ -97,54 +113,74 @@ public static int provisionTopics(final KafkaApiSpec apiSpec, final Admin adminC
 
+        final var statusList = new ArrayList<SchemaStatus>();
+
         domainTopics.forEach(
                 (topic -> {
+                    final var schemaSubject = topic.name() + "-value";
                     final var schemaInfo = apiSpec.schemaInfoForTopic(topic.name());
                     final var schemaRef = schemaInfo.schemaRef();
-                    final Path schemaPath = Paths.get(baseResourcePath, schemaRef);
+                    final var schemaPath = Paths.get(baseResourcePath, schemaRef);
+                    final var status =
+                            SchemaStatus.builder()
+                                    .schemaSubject(schemaSubject)
+                                    .schemaInfo(schemaInfo)
+                                    .schemaPath(schemaPath.toAbsolutePath().toString());
+
                     final String schemaContent;
                     try {
                         schemaContent = readString(schemaPath, UTF_8);
                     } catch (IOException e) {
+                        status.exception(e);
+                        statusList.add(status.build());
                         throw new ProvisioningException(
                                 "Failed to read schema from: " + schemaPath, e);
                     }
-                    final ParsedSchema someSchema =
-                            getSchema(topic.name(), schemaRef, baseResourcePath, schemaContent);
 
                     // register the schema against the topic (subject)
                     try {
-                        schemaRegistryClient.register(topic.name() + "-value", someSchema);
+                        final ParsedSchema someSchema =
+                                getSchema(topic.name(), schemaRef, baseResourcePath, schemaContent);
+
+                        schemaRegistryClient.register(schemaSubject, someSchema);
                     } catch (IOException | RestClientException e) {
+                        status.exception(e);
+                        statusList.add(status.build());
                         throw new ProvisioningException(
                                 "Failed to register schema. topic: "
                                         + topic.name()
                                         + ", schema:"
                                         + schemaPath,
                                 e);
                     }
+                    statusList.add(status.build());
                 }));
+        return statusList;
     }
 
     /**
      * Provision acls in the Kafka cluster
      *
      * @param apiSpec the api spec.
      * @param adminClient th admin client for the cluster.
+     * @return status of provisioning
      * @throws ProvisioningException on interrupt
      */
-    public static void provisionAcls(final KafkaApiSpec apiSpec, final Admin adminClient) {
+    public static AclStatus provisionAcls(final KafkaApiSpec apiSpec, final Admin adminClient) {
```

Comment:
Do you not want `AclStatus` to include the concept of failure like `SchemaStatus`?